server/src/main/java/com/datastrato/gravitino/server/web/ui/WebUIFilter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.server.web.ui;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+// This filter is used to serve static HTML files from the Gravitino WEB UI.
+// https://nextjs.org/docs/pages/building-your-application/deploying/static-exports#deploying
+public class WebUIFilter implements Filter {
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) request;
+    String path = httpRequest.getRequestURI();
+    if (path.equals("/")) {
+      // Redirect to the index page.
+      httpRequest.getRequestDispatcher("/ui/index.html").forward(request, response);
+    } else if (path.startsWith("/ui/") && !path.contains(".")) {
```

Comment:
What is the meaning of `!path.contains(".")`?

Reply:
Do we have a scenario like "/ui/xxx/", which is ended with "/"?

Reply:
> What is the meaning of !path.contains(".")?
Skips `*.js`, `*.css`, and more.